### PR TITLE
Speedometer minor updates

### LINKED ISSUE #21: Display speedometer in App
**User Story**

As an app user, I want to see the travel speed of my car, so that I can ensure the car is adhering to the traffic rules.

**Acceptance Criteria**

- The app displays the the speed of the car with frequent updates.
- The speed of the car will be accurately displayed.

### PR DESCRIPTION
**Pull request by Aditya-Nair-gu**
Implemented speedometer using  DistanceCar as suggested by @platisd in closed pull request #27

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -27,28 +26,15 @@ const int lDegrees = -75; // degrees to turn left
...
-
-
-
 ArduinoRuntime arduinoRuntime;
 BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
 BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
 DifferentialControl control(leftMotor, rightMotor);
 SimpleCar car(control);
-    
+DirectionlessOdometer odometer(arduinoRuntime, smartcarlib::pins::v2::leftOdometerPin, []() { odometer.update(); }, 100);
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino]: Why not two odometers?

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -27,28 +26,15 @@ const int lDegrees = -75; // degrees to turn left
...
 bool flag = false;
 
-
-
-
 ArduinoRuntime arduinoRuntime;
 BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
 BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
 DifferentialControl control(leftMotor, rightMotor);
 SimpleCar car(control);
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino]: Why is the `SimpleCar` instance still here?

**Review by platisd — CHANGES_REQUESTED** [review]

**Comment by platisd** [comment]
Also the pull request does not have:
>  Meaningful title and valuable description in pull request

**Comment by platisd** [comment]
You have not addressed all comments yet

File: arduino/smartcar/smartcar.ino [hunk 3]
```diff
@@ -17,28 +17,20 @@ char pass[] = " ";
...
 ArduinoRuntime arduinoRuntime;
 BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
 BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
 DifferentialControl control(leftMotor, rightMotor);
-SimpleCar car(control);
-    
-SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
-std::vector<char> frameBuffer;
-
+//SimpleCar car(control);
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino]: Why is this not removed?

File: arduino/smartcar/smartcar.ino [hunk 4]
```diff
@@ -47,34 +39,36 @@ DirectionlessOdometer rightOdometer{ arduinoRuntime,
                                      smartcarlib::pins::v2::rightOdometerPin,
                                      []() { rightOdometer.update(); },
                                      pulsesPerMeter };
-
-
+ 
+DistanceCar car1(arduinoRuntime,control, leftOdometer, rightOdometer);
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino]: why `car1` and not `car`?

**Review by platisd — CHANGES_REQUESTED** [review]

**Comment by platisd** [comment]
You shouldn't be the ones that fill out the entire DoD checklist :sweat_smile:

**Comment by Aditya-Nair-gu** [comment]
> You shouldn't be the ones that fill out the entire DoD checklist 😅

We don't check the last 2  check boxes? @platisd

File: arduino/smartcar/smartcar.ino (hunk 3 above)
Comment (platisd) [reply arduino/smartcar/smartcar.ino]: This is not resolved. Why is there commented out code and not removed?

File: arduino/smartcar/smartcar.ino [hunk 5]
```diff
@@ -143,16 +137,11 @@ void loop()
...
-        const auto avgOdometerSpeed = String((leftOdometerSpeed + rightOdometerSpeed)/2);
-        //std::string s = std::to_string(avgOdometerSpeed);
-        //const char* b = s.c_str();
+        const auto avgOdometerSpeed = String(car.getSpeed());
         mqtt.publish("/smartcar/speedometer",  avgOdometerSpeed); 
     }
    }
     Serial.println(front.getDistance());
-    Serial.println(leftOdometer.getDistance());
+   // Serial.println(leftOdometer.getDistance());
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino]: Why is there commented out code and not removed?

**Comment by platisd** [comment]
> > You shouldn't be the ones that fill out the entire DoD checklist sweat_smile
> 
> We don't check the last 2 check boxes? @platisd

You should check them when they are **true**. For example, how can you check the box about comprehending the changes when there has been no review yet? Or why would the final one about the state of the target branch be checked when someone's requesting changes?

**Review by platisd — APPROVED** [review]

**Pull request merged**